packages/astro/src/core/endpoint/dev/index.ts

```diff
@@ -18,5 +26,11 @@ export async function call(options: SSROptions, logging: LogOptions) {
 		route: options.route,
 	});
 
-	return await callEndpoint(endpointHandler, env, ctx, logging);
+	return await callEndpoint(
+		endpointHandler,
+		env,
+		ctx,
+		logging,
+		middleware as AstroMiddlewareInstance<Response | EndpointOutput>
```

Comment:
Could we avoid the cast if SSROptions.middleware was typed to this? Currently it is typed as `AstroMiddlewareInstance<unknown>`.